src/main/java/com/mixpanel/android/mpmetrics/MixpanelAPI.java

```diff
@@ -1691,6 +1727,12 @@ public void clearPushRegistrationId() {
         }
 
         @Override
+        public void clearSinglePushRegistrationId(String registrationId) {
+            remove("$android_devices", registrationId);
+            mPersistentIdentity.clearPushId();
```

Comment:
Why would we want to clear the current push ID as well? Current one may not be `registrationId`. Can we check whether is the same one or not before clearing the shared preferences? https://github.com/mixpanel/mixpanel-android/blob/master/src/main/java/com/mixpanel/android/mpmetrics/MixpanelAPI.java#L1686